engine/tests/ElggCoreGetEntitiesTest.php

```diff
@@ -348,6 +348,38 @@ public function testElggAPIGettersTSPValidTypeValidSubtype() {
 	}
 
 	/**
+	 * Makes sure that ege* returns only entities of specified type,
+	 * when subtype in type_subtype_pairs set to an empty or false value,
+	 * or an array with a single false/null value.
+	 * See #8851
+	 */
+	public function testElggAPIGettersTSPReturnsOnlyEntitiesOfTypeForEmptyOrFalseSubtype() {
+		$types = $this->getRandomValidTypes(1);
+		$type = $types[0];
+
+		// Test no value subtype
+		$tests = array(
+			ELGG_ENTITIES_ANY_VALUE,
+			ELGG_ENTITIES_NO_VALUE,
+			false,
+			array(),
+			array(ELGG_ENTITIES_ANY_VALUE),
+			array(ELGG_ENTITIES_NO_VALUE),
+		);
+
+		foreach ($tests as $subtype) {
+			$pair = array($type => $subtype);
+			$options = array(
+				'type_subtype_pairs' => $pair,
+				'wheres' => array("(e.type != '{$type}')"),
```

Comment:
Then we want `=`, not `!=`. This test is making the API look a lot more competent than it is.